compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -418,14 +399,34 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
                 false
               }
             }
-            .map(_._2)
+            .map(pair => (pair._2, pair._1.pos))
 
           regularClasses ::: staticForwarderClasses
         }
 
-        for (tree <- clDefs) {
-          generatedJSAST(tree)
-          genIRFile(cunit, tree)
+        for ((classDef, pos) <- clDefs) {
+          try {
+            val hashedClassDef = Hashers.hashClassDef(classDef)
+            generatedJSAST(hashedClassDef)
+            genIRFile(cunit, hashedClassDef)
+          } catch {
```

Comment:
Does it make sense to move this into `genIRFile`? It already has a `try` around `ir.Serializers.serialize` (and IIUC that's the function that would actually throw).

Reply:
It's `hashClassDef` that throws already, because we cannot hash `Transient`s. So no, it has to be here.